File: r/raindrops/raindrops.R
raindrops <- function(number) {
  sounds <- c("Pling" = 3, "Plang" = 5, "Plong" = 7)
  drops <- sounds[number %% sounds == 0] |> names() |> paste(collapse = "")
  ifelse(nchar(drops) > 1, yes = drops, no = as.character(number))
}

File: swift/raindrops/Sources/Raindrops/Raindrops.swift
struct Raindrops {
    let sounds: String

    init(_ number: Int) {
        var sounds = ""

        if (number % 3 == 0) { sounds += "Pling" }
        if (number % 5 == 0) { sounds += "Plang" }
        if (number % 7 == 0) { sounds += "Plong" }
        if (sounds.isEmpty) { sounds = String(number) }

        self.sounds = sounds
    }
}
